### PR TITLE
add support for mac syntax for sed replace

### PR DESCRIPTION
**Pull request by luketn**

### CODE REVIEW

File: x86_64-config.sh [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+if [ "$OS" = "Darwin" ]; then
+  # macOS
+  sed -i '' 's/aarch64/x86_64/g' .config
+  sed -i '' 's/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/g' kernel.code-workspace
+else
+  # Linux and others
+  sed 's/aarch64/x86_64/g' -i .config
+  sed 's/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/g' -i kernel.code-workspace
+fi
 sed 's/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/g' -i kernel.code-workspace
```
Comment (Jett59) [inline x86_64-config.sh:16]: Or this?

**Review by Jett59 — CHANGES_REQUESTED** [review]

File: aarch64-config.sh [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+if [ "$OS" = "Darwin" ]; then
+  # macOS
+  sed -i '' 's/x86_64/aarch64/g' .config
+  sed -i '' 's/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/g' kernel.code-workspace
+else
+  # Linux and others
+  sed 's/x86_64/aarch64/g' -i .config
+  sed 's/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/g' -i kernel.code-workspace
+fi
 sed 's/"rust-analyzer.cargo.target": "x86_64-unknown-linux-gnu"/"rust-analyzer.cargo.target": "aarch64-unknown-linux-gnu"/g' -i kernel.code-workspace
```
[the remark this answers is not in the document]
Comment (luketn) [reply aarch64-config.sh:16]: Good point!

File: x86_64-config.sh (hunk 1 above)
Comment (luketn) [reply x86_64-config.sh]: done

File: aarch64-config.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (luketn) [reply aarch64-config.sh]: done

**Review by Jett59 — APPROVED** [review]

**Pull request merged**